plugins/pulp_docker/plugins/importers/sync.py

```diff
@@ -300,15 +303,21 @@ def _process_manifest(self, manifest, digest, available_blobs, tag=None):
             manifest_file.write(manifest)
         manifest = models.Manifest.from_json(manifest, digest)
         self.parent.available_manifests.append(manifest)
+        foreign_layer = 'application/vnd.docker.image.rootfs.foreign.diff.tar.gzip'
+        has_foreign_layer = False
         for layer in manifest.fs_layers:
-            available_blobs.add(layer.blob_sum)
+            if layer.layer_type and layer.layer_type == foreign_layer:
+                has_foreign_layer = True
+            else:
+                available_blobs.add(layer.blob_sum)
         if manifest.config_layer:
             available_blobs.add(manifest.config_layer)
         self.progress_successes += 1
         # Remember this tag for the SaveTagsStep.
         if tag:
             self.parent.save_tags_step.tagged_manifests.append((tag, manifest,
                                                                 constants.MANIFEST_IMAGE_TYPE))
+        return has_foreign_layer
```

Comment:
The method doc block disagrees about the return value. It looks like it was wrong to begin with, but it's worth changing it to match this behavior.